client/trino-cli/src/main/java/io/trino/cli/StatusPrinter.java

```diff
@@ -323,11 +323,24 @@ private void printQueryInfo(QueryStatusInfo results, WarningsPrinter warningsPri
             verify(terminalWidth >= 75); // otherwise handled above
             int progressWidth = (min(terminalWidth, 100) - 75) + 17; // progress bar is 17-42 characters wide
 
-            if (stats.isScheduled()) {
-                String progressBar = formatProgressBar(progressWidth,
-                        stats.getCompletedSplits(),
-                        max(0, stats.getRunningSplits()),
-                        stats.getTotalSplits());
+            if (stats.isFaultTolerantStageScheduled() || stats.isScheduled()) {
+                String progressBar;
+                if (stats.isFaultTolerantStageScheduled()) {
+                    int completed = stats.getCompletedSplits();
+                    int running = stats.getRunningSplits();
+                    int total = stats.getTotalSplits();
+                    if (progressPercentage > 0) {
+                        // prevent progress bar from going back and forth for fault tolerant execution
+                        total = max(total, (completed * 100 + progressPercentage - 1) / progressPercentage);
```

Comment:
I do not get this formula. Should we just plot the progress bar based on `progressPercentage` (if `progressPercentage` is 10% then we plot progress bar which ish 10% of total length?

Reply:
Yeah I know this may cause confusion.

Currently it's like this:
// 0:17 [ 103MB,  802K rows] [5.74MB/s, 44.9K rows/s] [=====>>                                   ] 10%
where "=" represents completed splits, ">" represents running splits. And 10% = #completed / #total.

With the new formula if we continue to use the same representation, then there will be inconsistency. `total = max(total, (completed * 100 + progressPercentage - 1) / progressPercentage);` ensures progress bar doesn't go back and forth (which may cause confusion).